### PR TITLE
Add explanations to the runtime attach readme

### PR DESCRIPTION
**Pull request by jeanbisutti**

### CODE REVIEW

File: runtime-attach/README.md [hunk 1]
```diff
@@ -10,10 +10,27 @@ The attachment will _not_ be initiated in the following cases:
...
+* The temporary directory should be writable
+
+To use runtime attachment, you have to add the following dependency:
+
+_Maven_
+```xml
+<dependency>
+  <groupId>io.opentelemetry.contrib</groupId>
+  <artifactId>opentelemetry-runtime-attach</artifactId>
+  <version>...</version>
```
Comment (jkwatson-verta) [inline runtime-attach/README.md:23]: don't we also need the agent itself in our dependencies? Or does runtime-attach already depend on the agent transitively?

File: runtime-attach/README.md [hunk 2]
```diff
@@ -10,10 +10,27 @@ The attachment will _not_ be initiated in the following cases:
 * The attachment is not requested from the _main_ thread
 * The attachment is not requested from the `public static void main(String[] args)` method
 * The agent is already attached
-* The application is running on a JRE (a JDK is necessary)
-* The application is running on a read-only file system
+* The application is running on a JRE (a JDK is necessary, see [this issue](https://github.com/raphw/byte-buddy/issues/374))
+* The temporary directory should be writable
+
+To use runtime attachment, you have to add the following dependency:
```
Comment (trask) [inline runtime-attach/README.md:16]: check out doc pattern used in instrumentation repo for documenting how to add dependencies: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/okhttp/okhttp-3.0/library/README.md#add-these-dependencies-to-your-project

File: runtime-attach/README.md [hunk 3]
```diff
@@ -10,10 +10,27 @@ The attachment will _not_ be initiated in the following cases:
...
+  <artifactId>opentelemetry-runtime-attach</artifactId>
+  <version>...</version>
+</dependency>
+```
+_Gradle_
+```
+implementation 'io.opentelemetry.contrib:opentelemetry-runtime-attach:...'
+```
+
+After, you have to request the attachment of the agent _at the beginning of the `public static void main(String[] args)` method_ with the help of the `io.opentelemetry.contrib.attach.RuntimeAttach.attachJavaagentToCurrentJVM()` method.
```
Comment (trask) [inline runtime-attach/README.md:31]: the italicized `public static void main` looks odd to me, maybe just italicize _at the beginning of_?

Comment (trask) [reply runtime-attach/README.md:31]: (separately, `attachJavaagentToCurrentJVM` seems very long, what do you think of proposing to change it to simply `attach` when moving it to instrumentation repo?)

File: runtime-attach/README.md (hunk 1 above)
Comment (jeanbisutti) [reply runtime-attach/README.md]: This dependency embeds the OpenTelemetry agent JAR. I have added a sentence to the readme.

**Review by mateuszrzeszutek — APPROVED** [review]

File: runtime-attach/README.md [hunk 4]
```diff
@@ -2,18 +2,37 @@
 
 If you can't update the JVM arguments to attach the [OpenTelemetry Java agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation) (_-javaagent:path/to/opentelemetry-javaagent.jar_), this project allows you to do attachment programmatically.
 
+## Quick start
+
+### Add a dependency
+
+Replace `OPENTELEMETRY_CONTRIB_VERSION` with the latest stable release.
```
[the remark this answers is not in the document]
Comment (jeanbisutti) [reply runtime-attach/README.md:9]: 😂

**Pull request merged**